server/src/main/java/org/elasticsearch/TransportVersion.java

```diff
@@ -229,20 +227,48 @@ public static TransportVersion max(TransportVersion version1, TransportVersion v
     }
 
     /**
-     * returns a sorted set of all transport version constants
+     * returns a sorted collection of declared transport version constants
      */
-    public static NavigableSet<TransportVersion> getAllVersions() {
-        return ALL_VERSIONS;
+    public static Collection<TransportVersion> getAllVersions() {
+        return VERSION_IDS.values();
     }
 
-    final int id;
+    public final int id;
     private final String uniqueId;
 
     TransportVersion(int id, String uniqueId) {
         this.id = id;
         this.uniqueId = Strings.requireNonEmpty(uniqueId, "Each TransportVersion needs a unique string id");
     }
 
+    /**
+     * Placeholder method for code compatibility with code calling {@code CURRENT.minimumCompatibilityVersion}.
+     */
+    @Deprecated
+    public TransportVersion minimumCompatibilityVersion() {
```

Comment:
This is the 'basic' compatibility method, replacing `Version.CURRENT.minimumCompatibilityVersion()`. Callers of this method can be replaced with `TransportVersion.MINIMUM_COMPATIBLE` in due course.

Reply:
I am happy with this plan but..
alternatively we could consider to keep these versions as they are in Version. This should work in on-prem, and once we clarify the handling of this method in serverless we would refactor it. The calculateMinimumCompatVersion could remain private as it is in Version.
This would possibly help us focus on refactoring usages of Version->TransportVersion. The changes would be more mechanical and easier to review.

Reply:
Moving references to a specific method as part of a first pass through all the code helps us differentiate between transport-related uses of `minimumCompatibleVersion`, which we need to refactor, and other uses, which we need to ignore for now.